src/pytorch/neighbors/getNeighborPairsCUDA.cu

```diff
@@ -151,6 +166,13 @@ public:
                 get_accessor<scalar_t, 2>(deltas),
                 get_accessor<scalar_t, 1>(distances),
                 get_accessor<scalar_t, 2>(box_vectors));
+	    cudaEventRecord(event, stream);
+	    cudaEventSynchronize(event);
```

Comment:
This may hurt performance. CPU will be waiting for the neighbor kernel to finish rather than working on dispatching the next kernel.

Reply:
This will definitely hurt performance, but it is the only way to get the value from the error flag before this function exits. I put it in managed memory, but a synchronization barrier is needed anyway. I believe an event like this is the less invasive form of sync in this case.
The sync might be delayed until line 158 here:
https://github.com/openmm/NNPOps/blob/8b2d427888e548dbb655b6233060e3e8038e92c2/src/pytorch/neighbors/getNeighborPairsCUDA.cu#L156-L159

If the function must guarantee to throw I cannot think of any other way.